src/lib.rs

```diff
@@ -54,10 +54,10 @@ struct CResult<T: Copy> {
     error: CDisposableString,
 }
 
-impl<T: Copy> Into<Result<T, String>> for CResult<T> {
+impl<T: Copy, V: Into<T> + Copy> Into<Result<T, String>> for CResult<V> {
     fn into(self) -> Result<T, String> {
         if self.error.value.is_null() {
-            Ok(self.value)
+            Ok(self.value.into())
```

Comment:
Unnecessary change, but I think this makes sense?

Reply:
Thank you for the PR on the first place.

I follow yagni principle so I think we don't need to add it until it's not required somewhere.